packages/babel-helper-create-class-features-plugin/src/index.js

```diff
@@ -49,141 +51,158 @@ export function createClassFeaturePlugin({
     },
 
     visitor: {
-      Class(path, state) {
-        if (this.file.get(versionKey) !== version) return;
+      Class: {
+        enter(path, state) {
+          if (this.file.get(versionKey) !== version) return;
 
-        verifyUsedFeatures(path, this.file);
-
-        const loose = isLoose(this.file, feature);
+          verifyUsedFeatures(path, this.file);
 
-        let constructor;
-        let isDecorated = hasOwnDecorators(path.node);
-        const props = [];
-        const elements = [];
-        const computedPaths = [];
-        const privateNames = new Set();
-        const body = path.get("body");
+          const loose = isLoose(this.file, feature);
+          const privateNamesMap = new Map(
+            privateNamesStack[privateNamesStack.length - 1],
```

Comment:
Since nodes can be re-queued and skipped, it's not guaranteed that they are visited in-order.

Instead of a stack, I think we could use a `WeakMap` from class nodes to `privateNamesMap`, and get the parent class using `path.findParent(p => p.isClass())`.